File: app/src/main/java/com/yashwant/newsmodule/app/Utility.java
package com.yashwant.newsmodule.app;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by yashwant on 18/10/17.
 */

public class Utility {
    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivityManager
                = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }

    public static Date formatedDate(String dateStr) throws ParseException {

        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
        Date date = null;
        try {
             date= format.parse(dateStr);
            System.out.println(date);
        } catch (ParseException e) {
            e.printStackTrace();
        }


        return date;


    }

}

File: app/src/main/java/com/yashwant/newsmodule/app/Utility.kt
package com.yashwant.newsmodule.app

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date

/**
 * Created by yashwant on 18/10/17.
 */

object Utility {
    fun isNetworkAvailable(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetworkInfo = connectivityManager.activeNetworkInfo
        return activeNetworkInfo != null && activeNetworkInfo.isConnected
    }

    @Throws(ParseException::class)
    fun formatedDate(dateStr: String): Date? {

        val format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
        var date: Date? = null
        try {
            date = format.parse(dateStr)
            println(date)
        } catch (e: ParseException) {
            e.printStackTrace()
        }


        return date


    }

}
